smartsim/_core/entrypoints/dragon.py

```diff
@@ -90,39 +95,63 @@ def print_summary(network_interface: str, ip_address: str) -> None:
                 HOSTNAME: {socket.gethostname()}
                 DRAGON_SERVER_CONFIG: {json.dumps(zmq_config)}
                 --------------------------------------
+                """),
+        )
 
-                --------------- Output ---------------
 
-                """),
+def restart_updater(
+    backend: DragonBackend, updater: t.Optional[ContextThread]
+) -> ContextThread:
+    # If the updater was started, check if it completed or died
+    if updater is not None:
+        updater.join(0.1)
+        # If it's alive, there is nothing to do
+        if updater.is_alive():
+            return updater
+    updater = ContextThread(name="DragonBackend", daemon=True, target=backend.update)
+    updater.start()
+    return updater
+
+
+def is_updater_healthy(backend: DragonBackend) -> bool:
+
+    heartbeat_delay = backend.current_time - backend.last_heartbeat
+    if heartbeat_delay > 30.0 + float(backend.cooldown_period):
```

Comment:
- Can the 30 here be rolled into `backend.cooldown_period` since it's fixed? 
- If no above, should we also make the 30 configuration based or does 30 have special meaning only in this context?
  - if there is a special meaning/constraint to 30, can we document it?

Reply:
Yeah, totally, I should make it related to the server timeout somehow.